.github/workflows/on-demand-cms-sync.yml

```
@@ -0,0 +1,67 @@
+name: Sync Squidex data to Contentful
+on:
+  workflow_dispatch:
+    inputs:
+      environment-name:
+        required: true
+        description: The environment name
+        type: string
+      entity:
+        required: true
+        type: string
+        description: Choose which Contentful entity (or entities) to sync
+        default: all
+      contentful-env:
+        required: true
+        type: string
+        description: Choose which Contentful environment to sync to
+      squidex-app:
+        required: true
+        type: string
+        description: Choose which Squidex app to sync from
+
+jobs:
+  sync_cms_data:
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.environment-name }}
+    container:
+      image: ghcr.io/yldio/asap-hub/node-python-sq:7b77d99657ab3718ed548ba366ffbcbb15315fd8
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup
+        id: setup
+        uses: ./.github/actions/setup-environment
+        with:
+          environment-name: ${{ inputs.environment-name }}
+      - name: Run CMS sync for news
+        if: ${{ inputs.entity == 'news' || inputs.entity == 'all' }}
+        run: |
+          yarn install
+          yarn build:babel
+          yarn build:typecheck
+          yarn workspace @asap-hub/cms-data-sync migrate:news
```

You can try this but you'll need this workflow to be dependant on "build"

```suggestion
            - name: Cache build output
        uses: ./.github/actions/cache-unplugged
        with:
          environment-name: ${{ inputs.environment-name }}
      - name: Cache build output
        uses: ./.github/actions/cache-unplugged
        with:
          environment-name: ${{ inputs.environment-name }}
      - name: Cache build typecheck output
        uses: ./.github/actions/cache-build
        with:
          cache-prefix: ${{ steps.setup.outputs.cache-prefix-definitions }}
          environment-name: ${{ inputs.environment-name }}
      - name: Cache transpile output
        uses: ./.github/actions/cache-build
        with:
          cache-prefix: ${{ steps.setup.outputs.cache-prefix-transpile }}
          environment-name: ${{ inputs.environment-name }}

      - name: Run CMS sync for news
        if: ${{ inputs.entity == 'news' || inputs.entity == 'all' }}
        run: |
          yarn workspace @asap-hub/cms-data-sync migrate:news
```